### PR TITLE
[Launch] add gpu report during training

### PR DESCRIPTION
**Pull request by kuizhiqing**
### PR types

New features

### PR changes

Others

### Describe

* add nvsmi util for gpu
* add watcher to collect gpu info

### CODE REVIEW

File: python/paddle/distributed/launch/controllers/controller.py [hunk 1]
```diff
@@ -39,6 +40,8 @@ def __init__(self, ctx):
         self.ctx = ctx
         self.master = Master.factory(self.ctx)
 
+        self.watcher = Watcher(self.ctx)
```
Comment (xymyeah) [inline python/paddle/distributed/launch/controllers/controller.py:43]: 是否需要有开关？用户可以关掉？

File: python/paddle/distributed/launch/controllers/watcher.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        if len(self.gpus) > 0:
+            fn = os.path.join(self.ctx.args.log_dir,
+                              "{}.gpu.log".format(self.ctx.args.job_id))
+            os.makedirs(os.path.dirname(fn), exist_ok=True)
+            self.gpu_fd = open(fn, 'w')
+        else:
+            return
+
+        # start
+        self.proc = Thread(target=self.watch)
```
Comment (xymyeah) [inline python/paddle/distributed/launch/controllers/watcher.py:41]: Daemon thread？Multi Threads or Multi Process ？

File: python/paddle/distributed/launch/utils/nvsmi.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+# limitations under the License.
+
+import subprocess
+import shlex
+import os
+import json
+import shutil
+
+
+class Info(object):
```
Comment (xymyeah) [inline python/paddle/distributed/launch/utils/nvsmi.py:22]: 目前只支持GPU，DCU、XPU、NPU是否要支持？

**Review by xymyeah — CHANGES_REQUESTED** [review]

File: python/paddle/distributed/launch/controllers/controller.py (hunk 1 above)
Comment (kuizhiqing) [reply python/paddle/distributed/launch/controllers/controller.py:43]: 目前是默认存在 nvidia-smi 即开启，不存在则关闭。独立线程不会对用户造成影响。

File: python/paddle/distributed/launch/controllers/watcher.py (hunk 2 above)
Comment (kuizhiqing) [reply python/paddle/distributed/launch/controllers/watcher.py:41]: The exiting logic is handled so non-daemon is OK, while we can simply set daemon true.
Multiprocess have issue with handling system signal, thread is preferred.

File: python/paddle/distributed/launch/utils/nvsmi.py (hunk 3 above)
Comment (kuizhiqing) [reply python/paddle/distributed/launch/utils/nvsmi.py:22]: 先支持 gpu 吧，后面整体稳定、落地之后迁移应该很快。

**Review by xymyeah — APPROVED** [review]
LGTM

**Review by ForFishes — APPROVED** [review]
LGTM

**Pull request merged**